wandb/integration/ultralytics/callback.py

```diff
@@ -0,0 +1,367 @@
+        visualize_skeleton: visualize pose skeleton by drawing lines connecting
+            keypoints for human pose.
+    """
+
+    def __init__(
+        self,
+        model: YOLO,
+        max_validation_batches: int = 1,
+        enable_model_checkpointing: bool = False,
+        visualize_skeleton: bool = False,
+    ) -> None:
+        self.max_validation_batches = max_validation_batches
+        self.enable_model_checkpointing = enable_model_checkpointing
+        self.visualize_skeleton = visualize_skeleton
+        self._make_tables(model)
+        self._make_predictor(model)
+
+    def _make_tables(self, model: YOLO):
+        if model.task == "detect":
+            validation_columns = [
+                "Data-Index",
+                "Batch-Index",
+                "Image",
+                "Mean-Confidence",
+                "Speed",
+            ]
+            train_columns = ["Epoch"] + validation_columns
+            self.train_validation_table = wandb.Table(columns=train_columns)
+            self.validation_table = wandb.Table(columns=validation_columns)
+            self.prediction_table = wandb.Table(
+                columns=["Image", "Num-Objects", "Mean-Confidence", "Speed"]
+            )
+        elif model.task == "classify":
+            classification_columns = [
+                "Image",
+                "Predicted-Category",
+                "Prediction-Confidence",
+                "Top-5-Prediction-Categories",
+                "Top-5-Prediction-Confindence",
+                "Probabilities",
+                "Speed",
+            ]
+            validation_columns = ["Data-Index", "Batch-Index"] + classification_columns
+            validation_columns.insert(3, "Ground-Truth-Category")
+            train_columns = ["Epoch"] + validation_columns
+            self.train_validation_table = wandb.Table(columns=train_columns)
+            self.validation_table = wandb.Table(columns=validation_columns)
+            self.prediction_table = wandb.Table(columns=classification_columns)
+        elif model.task == "pose":
+            validation_columns = [
+                "Data-Index",
+                "Batch-Index",
+                "Image-Ground-Truth",
+                "Image-Prediction",
+                "Num-Instances",
+                "Mean-Confidence",
+                "Speed",
+            ]
+            train_columns = ["Epoch"] + validation_columns
+            self.train_validation_table = wandb.Table(columns=train_columns)
+            self.validation_table = wandb.Table(columns=validation_columns)
+            self.prediction_table = wandb.Table(
+                columns=[
+                    "Image-Prediction",
+                    "Num-Instances",
+                    "Mean-Confidence",
+                    "Speed",
+                ]
+            )
+
+    def _make_predictor(self, model: YOLO):
+        overrides = model.overrides.copy()
```

Comment:
Consider using `copy.deepcopy` as `copy` still creates only a shallow copy and modifies the original object upon modification to the copied object.